test/Test.hs

```diff
@@ -226,7 +228,7 @@ main = defaultMain $
 
 #ifdef MIN_VERSION_tasty_expected_failure
       , expectFail $ testCase "nontermination even when fair" $
-        [[2]] @=? observeManyT 2 oddsOrTwo
+        (Right [2] @=?) =<< (nonTerminating $ observeManyT 2 oddsOrTwo)
```

Comment:
Do we need `expectFail` any longer? We could just test that `nonTerminating $ ...` return `Left ()`, right?

Reply:
We could, but my hope was that in the future the tasty + tasty-expected-failure timeout handling could be approved and the `nonTerminating` could be dropped.  I also like that it indicates this is a problematic scenario (although admittedly that could also be done with just a comment).

Reply:
I see your point, but even if `tasty` timeout story improves in future releases, I'd prefer to avoid tightening lower bounds of `tasty` and `tasty-expected-failures`. Stack users are already forced to use `extra-deps`, because `lts-16` features only `tasty-expected-failure-0.11.1.2`, and I'd rather make their life easier than worse. 

`testCase $ (Left () @=?) =<< nonTerminating ... ` would allow us to shave off CPP here and in the cabal file.

Reply:
That sounds reasonable, let me make that adjustment.